save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ProjectService.kt

```diff
@@ -135,45 +139,49 @@ class ProjectService(
     fun getAllAsFluxByOrganizationName(organizationName: String) = getAllByOrganizationName(organizationName).let { Flux.fromIterable(it) }
 
     /**
-     * @return project's without status
-     */
-    fun getNotDeletedProjects(): List<Project> {
-        val projects = projectRepository.findAll { root, _, cb ->
-            cb.notEqual(root.get<String>("status"), ProjectStatus.DELETED)
-        }
-        return projects
-    }
-
-    /**
-     * @param organizationName
+     * @param organizationName is [organization] name
      * @param authentication
+     * @param statuses is status`s
      * @return list of not deleted projects
      */
-    fun getNotDeletedProjectsByOrganizationName(
+    fun getProjectsByOrganizationNameAndStatusIn(
         organizationName: String,
         authentication: Authentication?,
+        statuses: Set<ProjectStatus> = setOf(ProjectStatus.CREATED)
     ): Flux<Project> = getAllAsFluxByOrganizationName(organizationName)
         .filter {
-            it.status != ProjectStatus.DELETED
+            it.status in statuses
         }
         .filter {
             projectPermissionEvaluator.hasPermission(authentication, it, Permission.READ)
         }
 
     /**
-     * @param projectFilters
+     * @param projectFilters is filter for [projects]
      * @return project's with filter
      */
-    fun getNotDeletedProjectsWithFilter(projectFilters: ProjectFilters?): List<Project> {
-        val name = projectFilters?.name?.let { "%$it%" }
-        val projects = projectRepository.findAll { root, _, cb ->
-            val namePredicate = name?.let { cb.like(root.get("name"), it) } ?: cb.and()
-            cb.and(
-                namePredicate,
-                cb.notEqual(root.get<String>("status"), ProjectStatus.DELETED)
-            )
-        }
-        return projects
+    fun getFiltered(projectFilters: ProjectFilters): List<Project> =
+            if (projectFilters.organizationName.isBlank()) {
```

Comment:
what's about using `when` here?
```suggestion
when (projectFilters.organizationName.isBlank() to projectFilters.name.isBlank()) {
    true to true -> projectRepository.findByStatusIn(projectFilters.statuses)
    true to false -> projectRepository.findByNameLikeAndStatusIn(wrapValue(projectFilters.name), projectFilters.statuses)
    false to true -> projectRepository.findByOrganizationNameAndStatusIn(projectFilters.organizationName, projectFilters.statuses)
    false to false -> findByNameAndOrganizationNameAndStatusIn(projectFilters.name, projectFilters.organizationName, projectFilters.statuses)?.let { listOf(it) }.orEmpty()
    else -> throw IllegalStateException("Impossible state")
}
```